src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -45,6 +45,9 @@ public CommandResult execute(String commandText) throws CommandException, ParseE
 
         CommandResult commandResult;
         Command command = addressBookParser.parseCommand(commandText);
+        if (command.isUndoable()) {
+            model.addToHistory();
+        }
```

Comment:
may be good for regular commands to not have an undoable field and
something like if instanceof UndoableCommand instead